File: 2/2864/MaximumOddBinaryNumber.ts
/*
https://leetcode.com/problems/maximum-odd-binary-number/submissions/1190581163/?envType=daily-question&envId=2024-03-01
*/

function maximumOddBinaryNumber(s: string): string {
    let countOnes:number=0, countZeros:number=0, sz:number=s.length;
    for(let i:number=0; i<sz; ++i){
        if(s[i]=='0'){
            countZeros++;
        } else countOnes++;
    }
    let res:string="";
    while(countOnes!=1){
        res+="1";
        countOnes--;
    }
    while(countZeros!=0){
        res+="0";
        countZeros--;
    }
    res+="1";
    return res;
};

File: 2/2864/MaximumOddBinaryNumber.py
# https://leetcode.com/problems/maximum-odd-binary-number/description/?envType=daily-question&envId=2024-03-01

class Solution:
    def maximumOddBinaryNumber(self, s: str) -> str:
        countOnes=0
        countZeros=0
        sz=len(s)
        for i in range(0, sz, 1):
            if s[i]=='0':
                countZeros+=1
            else:
                countOnes+=1
        res=""
        while(countOnes!=1):
            res=res+"1"
            countOnes-=1
        while(countZeros!=0):
            res=res+"0"
            countZeros-=1
        res=res+"1"
        return res